### PR TITLE
1327220: Limit IN queries to no more than 500 {hypervisor, guest}Ids

### PR DESCRIPTION
**Pull request by cnsnyder**
This fix is the same as the one that was backported to candlepin-0.9.49-HOTFIX in commit: 45b05945ceb63c541354ac8c036e460c50bb587e

### CODE REVIEW

**Comment by vritant** [comment]
needs #1190

**Comment by awood** [comment]
Please fix the commit message on this.

The first line should not be longer than about 80 characters.  If the message gets broken into the comment text area in GitHub, then it is too long.

**Comment by cnsnyder** [comment]
No need for review yet. I think there might be a better way to do this.

File: server/src/main/java/org/candlepin/model/ConsumerCurator.java [hunk 1]
```diff
@@ -211,9 +213,22 @@ public VirtConsumerMap getGuestConsumersMap(Owner owner,
...
+            if (toIndex > possibleGuestIds.size()) {
+                toIndex = possibleGuestIds.size();
+            }
+            List<String> subList = possibleGuestIds.subList(fromIndex, toIndex);
+            q.setParameterList("guestids", subList);
+            q.setParameter("ownerid", owner.getId());
+            consumerUuids.addAll(q.list());
+            fromIndex = toIndex;
+            toIndex += MAX_IN_QUERY_LENGTH;
+        }
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/model/ConsumerCurator.java]: Pretty sure we have a standardized way of doing this in AbstractHibernateCurator

**Comment by Ceiu** [comment]
ACK

candlepin-0.9.54-HOTFIX:
```
curl -k -X POST -T ~/huge_files/hypervisor_guest_report.json -u admin:admin "https://localhost:8443/candlepin/hypervisors?owner=test"
{
  "displayMessage" : "Runtime Error rollback failed at org.postgresql.jdbc2.AbstractJdbc2Connection.checkClosed:714",
  "requestUuid" : "4d2a8090-8003-40df-924a-7f4665e33f0b"
}
```

PR:
```
...
    }, {
      "id" : "8a8d01cb542fc44201542fc4c18e426c",
      "guestId" : "e3ffd59c-6e85-4577-9600-bc6efcedd3ea",
      "created" : "2016-04-19T18:25:04.398+0000",
      "updated" : "2016-04-19T18:25:04.398+0000"
    }, {
(repeats a lot)
    "capabilities" : null,
    "hypervisorId" : {
      "id" : "8a8d01cb542fc44201542fc4ab5c0005",
      "hypervisorId" : "afea8305-bd5b-43b1-8b5d-3a6ae38b4a05",
      "created" : "2016-04-19T18:24:58.716+0000",
      "updated" : "2016-04-19T18:24:58.716+0000"
    },
    "contentTags" : null,
    "autoheal" : true,
    "dev" : false,
    "href" : "/consumers/91cd5f9a-3929-496e-8a43-f84ca629f495",
    "created" : "2016-04-19T18:24:58.710+0000",
    "updated" : "2016-04-19T18:25:04.468+0000"
  } ],
  "updated" : [ ],
  "unchanged" : [ ],
  "failedUpdate" : [ ]
}
```

**Pull request merged**